runtime/eventlog.c

```diff
@@ -69,13 +69,37 @@ struct event {
   uint64_t count; /* for misc counters */
 };
 
-#define EVENT_BUF_SIZE 4096
+/* events are stored thread-locally in event_buffers,
+   which are registered in a global doubly-linked list,
```

Comment:
If this is only for the purposes of flushing on exit, wouldn't Caml_state be a better place to put them? The cost of iterating Max_domains is relatively small if we're only doing it once at exit.